packages/oas-utils/src/helpers/redirect-to-proxy.ts

```diff
@@ -1,25 +1,37 @@
 import { isLocalUrl } from './is-local-url.ts'
 import { REGEX } from './regex-helpers.ts'
 
-/** Redirects the request to a proxy server with a given URL. */
+/**
+ * Redirects the request to a proxy server with a given URL. But not for:
+ *
+ * - Relative URLs
+ * - URLs that seem to point to a local IP
+ * - URLs that don't look like a domain
+ **/
 export function redirectToProxy(proxy?: string, url?: string): string {
-  if (!shouldUseProxy(proxy, url)) {
-    return url ?? ''
-  }
+  try {
+    if (!shouldUseProxy(proxy, url)) {
+      return url ?? ''
+    }
 
-  // Create new URL object from url
-  const newUrl = new URL(url as string)
+    // Create new URL object from url
+    const newUrl = new URL(url as string)
```

Comment:
```suggestion
    const newUrl = new URL(url)
```
why the type cast here, just do an early return above to check for undefined

Reply:
actually we can turn `shouldUseProxy` into a type guard for proxy then just check for url

Reply:
ahh, type guarding works for one parameter only, but still a great idea